tests/components/unifiprotect/conftest.py

```diff
@@ -160,3 +161,16 @@ async def enable_entity(
     await hass.async_block_till_done()
 
     return updated_entity
+
+
+def assert_entity_counts(
+    hass: HomeAssistant, platform: Platform, total: int, enabled: int
+) -> None:
+    """Assert entity counts for a given platform."""
+
+    entity_registry = er.async_get(hass)
+
+    entities = [e for e in entity_registry.entities if e.startswith(platform.value)]
```

Comment:
We have a `from homeassistant.core import split_entity_id` call that can be used here

Reply:
Like this?

```
entities = [e for e in entity_registry.entities if split_entity_id(e)[0] == platform.value]
```

I think it ends up being a wash. If anything, it might be a bet slow (the strings need to be scanned twice each instead of once).

Reply:
Yes, but it won't break if we ever have platforms are substrings of other platforms

Reply:
That is a fair point.